docs/current_docs/manuals/user/remotes/remote-repositories.mdx

```diff
@@ -4,10 +4,137 @@ slug: /manuals/user/remote-repositories
 
 # Remote Repositories
 
+Dagger supports the use of remote repositories in two distinct ways:
+
+- As directory arguments to Dagger Functions
+- As Dagger modules
+
+In both cases, Dagger supports the use of HTTP and SSH protocols for accessing remote repositories, compatible with all major Git hosting platforms such as GitHub, GitLab, BitBucket, Azure DevOps, Codeberg, and Sourcehut. For SSH references ("refs"), Dagger employs a unified authentication approach.
+
+## Remote repositories as directory arguments
+
 Dagger Functions that accept directory arguments are equally comfortable receiving local filesystem paths or remote Git repository references. In both cases, the CLI will convert it to an object referencing the contents of that filesystem path or Git repository location, and pass the resulting `Directory` object as argument to the Dagger Function.
 
-Here is an example of a Go builder Dagger Function that accepts a remote Git address (the address of Dagger's open-source repository) as a `Directory` argument, builds a binary version of the Dagger CLI and exports the build directory to the host filesystem as `./build`:
+### Supported protocols and schemes
+
+Dagger supports the following reference schemes for directory arguments:
+
+| Protocol | Scheme     | Public repositories | Private repositories | Example |
+|----------|------------|---------------------|----------------------|---------|
+| HTTP(S)  | Git HTTP   | Supported | In progress | `https://github.com/username/repo.git[#version[:subdir]]` |
```

Comment:
We discussed whether to document this at all.
- Argument against: The dir ref scheme is different from the module ref scheme and may change. If we document it, it's likely people will use it and things may break if we change it in future.
- Argument for: If we don't document it, people will assume that the dir ref scheme is the same as the module ref scheme, since that will be the only one documentated. This will however not work for dir args, leading to user frustration/a bad experience/help tickets in Discord.
- Second argument for: This is currently not documented, which means the only way to figure out the format for passing a remote repository arg is by looking for existing examples in the docs, searching GitHub, or asking in discord
I'm strongly in favour of documenting it, for both the above reasons. But please lmk in case you disagree or have other thoughts on this @marcosnils @grouville @shykes